zerver/views/realm.py

```diff
@@ -85,4 +85,8 @@ def update_realm(request, user_profile, name=REQ(validator=check_string, default
         do_set_realm_message_editing(realm, allow_message_editing, message_content_edit_limit_seconds)
         data['allow_message_editing'] = allow_message_editing
         data['message_content_edit_limit_seconds'] = message_content_edit_limit_seconds
+    # A user input of 0 means "no message retention", for which the respective backend
+    # value is None.
+    if message_retention_days == 0:
+        do_set_realm_property(realm, 'message_retention_days', None)
```

Comment:
This will result in the property being changed twice.  I think you want to just do

```
if message_retention_days == 0:  
    message_retention_days = None
```
closer to the top of the function